### PR TITLE
Geode 5428: Refactored DurableClientTests due to massive duplication

### PR DESCRIPTION
**Pull request by mhansonp**
This code was not in a good state. So the code has been refactored to not only address issues
where the code was giving bad feedback, but to address the impressive structural duplication.

Things done:
-Extracted a bunch of methods
-Added new Test files for test subjects that should have been separate
-Split out a base class so there was no inheritance of test classes to do basic things.
-Reparented classes that were inheriting from test classes

Co-Authored-By: Galen O'Sullivan <gosullivan@pivotal.io>

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Comment by mhansonp** [comment]
@upthewaterspout @WireBaron @balesh2 Please review when you get the chance.

File: geode-cq/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/DurableClientTestCase.java [hunk 1]
```diff
@@ -729,6 +730,12 @@ public void run2() throws CacheException {
       }
     });
 
+    try {
```
Comment (mhansonp) [inline geode-cq/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/DurableClientTestCase.java]: There seems to be some delay with the registering of interest. I'm not fond of this wait, but there isn't a better way I saw. Nothing to check for that would provide good feedback. I am open to suggestions.

File: geode-cq/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/DurableClientSimpleDUnitTest.java [hunk 2]
```diff
@@ -390,81 +327,33 @@ public void testDurableClientMultipleServersOneLive() {
...
-        assertEquals("Events were" + listener.events, numberOfEntries, listener.events.size());
-      }
-    });
-
-    try {
-      Thread.sleep(10000);
-    } catch (InterruptedException ex) {
-      fail("interrupted", ex);
-    }
+    sleep(10000);
```
Comment (mhansonp) [inline geode-cq/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/DurableClientSimpleDUnitTest.java]: This is a magic timeout. Remove it and things fail. Add it in and things are happy. I have no idea what this is really affecting, but it is necessary to work properly.

**Comment by mhansonp** [comment]
@galen-pivotal @WireBaron @upthewaterspout @balesh2 Can you all review this PR?

File: geode-cq/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/DurableClientSimpleDUnitTest.java (hunk 2 above)
Comment (galen-pivotal) [reply geode-cq/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/DurableClientSimpleDUnitTest.java]: This test appears to be ignored. Also it fails on my machine (when run with IntelliJ). Maybe a good future task.

**Pull request merged**